src/components/Table/LGFilters.tsx

```diff
@@ -1,46 +1,108 @@
 import { LeafyGreenTableRow } from "@leafygreen-ui/table/new";
-import { TableFilterPopover } from "components/TablePopover";
+import {
+  TableFilterPopover,
+  TableSearchPopover,
+  TableSortIcon,
+} from "components/TablePopover";
 import { TreeDataEntry } from "components/TreeSelect";
 
 type TreeSelectFilterProps = {
   "data-cy"?: string;
   tData: TreeDataEntry[];
-  title: React.ReactNode;
+  onConfirm?: ({ id, value }: { id: string; value: string[] }) => void;
 };
 
 export const getColumnTreeSelectFilterProps = ({
   "data-cy": dataCy,
+  onConfirm = () => {},
   tData,
-  title,
 }: TreeSelectFilterProps) => ({
-  header: ({ column }) => {
-    // Only present options that appear in the table
-    const options = tData.filter(
-      ({ value }) => !!column.getFacetedUniqueValues().get(value)
-    );
-    return (
-      <>
-        {title}
+  meta: {
+    filterComponent: ({ column }) => {
+      const filteredOptions = tData.filter(
+        ({ value }) => !!column.getFacetedUniqueValues().get(value)
+      );
+      return (
         <TableFilterPopover
           value={column?.getFilterValue() ?? []}
-          options={options}
-          onConfirm={(value) => {
-            column.setFilterValue(value);
+          options={filteredOptions.length ? filteredOptions : tData}
+          onConfirm={(newValue) => {
+            column.setFilterValue(newValue);
+            onConfirm({ id: column.id, value: newValue });
           }}
           data-cy={dataCy}
         />
-      </>
-    );
+      );
+    },
   },
   filterFn: (
     row: LeafyGreenTableRow<any>,
     columnId: string,
     filterValue: string[]
   ) => {
-    // If no filter is specified, show all rows
+    // If no filter is specified, show all rows.
     if (!filterValue.length) {
       return true;
     }
     return filterValue.includes(row.getValue(columnId));
   },
 });
+
+type InputFilterProps = {
+  "data-cy"?: string;
+  onConfirm?: ({ id, value }: { id: string; value: string }) => void;
+};
+
+export const getColumnInputFilterProps = ({
+  "data-cy": dataCy,
+  onConfirm = () => {},
+}: InputFilterProps) => ({
+  meta: {
```

Comment:
Use of `meta` is pretty neat! Good find, will be helpful with not duplicating code!